defcon/plugins/endpoint.py

```diff
@@ -0,0 +1,102 @@
+"""DefCon Prometheus Alertmanager plugin."""
+import requests
+import logging
+
+from defcon.plugins import base
+
+
+class EndpointPlugin(base.Plugin):
+    """DefCon Endpoint plugin.
+
+    set an endpoint to get a defcon level.
+    The endpoint should have the info {...'defcon':[1-5],...}
+    Another endPoint can be setup to force the level
+
+    Config:
+    ```python
+    {
+        'title': 'Defcon Production',
+        'url': 'http://defcon.com/api/simple/production/', // Url to endpoint
+        'link_status': 'http://defcon.com/status/production/ // Url to endpoint info
+        'force_url': 'http://defcon.com/api/simple/force/, // Url to force endpoint
+    }
+    ```
+    """
+
+    def __init__(self, config=None):
+        """Create an instance of the plugin."""
+        super(EndpointPlugin, self).__init__(config)
+        if config is None:
+            config = {}
+
+        self.url = config.get('url')
+        self.force_url = config.get('force_url', 'http://defcon.com/status')
+        self.title = config.get('title', '')
+        self.link_status = config.get('link_status', '')
+
+    @property
+    def short_name(self):
+        """Return the short name."""
+        return 'endpoint'
+
+    @property
+    def name(self):
+        """Return the name."""
+        return 'endpoint'
+
+    @property
+    def description(self):
+        """Return the description."""
+        return 'Returns statuses based on endpoint.'
+
+    @property
+    def link(self):
+        """Return the link."""
+        return 'https://github.com/iksaif/defcon'
+
+    def _get_defcon_from_url(self, url):
+        r = requests.get(self.url)
+        r.raise_for_status()
+        return (r.json().get('defcon', []))
+
+    def statuses(self):
+        """Return the generated statuses."""
+        ret = {}
+
+        if self._config is None:
+            return ret
+
+        defcon = 0
+
+        if self.force_url is not None:
```

Comment:
I do not really understand what force_url is trying to do. Can't you get away with only self.url ? (make sure you can configure the defcon level when this is failing).

Reply:
Since settings.py for each component is read only at the init, we need to be able to force a state.
This is what force_url stand for.
What we need is listend to a bunch of endpoint and determine the level of each, but if the force flag is here, put the same level everywhere

Reply:
This has been abandoned, only one url, no force allowed.